Makefile

```diff
@@ -2,59 +2,80 @@ org_path="github.com/mesosphere"
 repo_path="${org_path}/etcd-mesos"
 mkfile_path	:= $(abspath $(lastword $(MAKEFILE_LIST)))
 current_dir	:= $(patsubst %/,%,$(dir $(mkfile_path)))
-GOPATH=${current_dir}/Godeps/_workspace
+
 DOCKER_ORG=mesosphere
 VERSION=0.1.3
 
-default: clean deps build
+default: clean build
 
+.PHONY: clean
+.SILENT: clean
 clean:
-	-rm bin/etcd-*
-
-deps:
-	rm -f ${GOPATH}/src/${repo_path}
-	mkdir -p ${GOPATH}/src/${org_path}
-	ln -s ${current_dir} ${GOPATH}/src/${repo_path}
+	@rm -rf bin
 
-build: bin/etcd-mesos-executor bin/etcd-mesos-scheduler bin/etcd-mesos-proxy bin/etcd
-
-run: clean bin/etcd-mesos-executor bin/etcd run-scheduler
+.PHONY: format
+.SILENT: format
+format:
+	echo "gofmt'ing..."
+	@govendor fmt +local
 
+.PHONY: bin
+.SILENT: bin
 bin:
-	-mkdir bin
-
-bin/etcd-mesos-scheduler: bin
-	go build -o bin/etcd-mesos-scheduler cmd/etcd-mesos-scheduler/app.go
-
-bin/etcd-mesos-executor: bin
-	go build -o bin/etcd-mesos-executor cmd/etcd-mesos-executor/app.go
-
-bin/etcd-mesos-proxy: bin
-	go build -o bin/etcd-mesos-proxy cmd/etcd-mesos-proxy/app.go
-
-bin/etcd: bin
-	git submodule init
-	git submodule update
+	@mkdir -p bin
+
+.PHONY: build
+build: build_scheduler build_executor build_proxy build_etcd
+
+.PHONY: build_scheduler
+.SILENT: build_scheduler
+build_scheduler: bin
+	echo "Building etcd-mesos-scheduler.."
+	go build -o bin/etcd-mesos-scheduler ./cmd/etcd-mesos-scheduler
+
+.PHONY: build_executor
+.SILENT: build_executor
+build_executor: bin
+	echo "Building etcd-mesos-executor.."
+	go build -o bin/etcd-mesos-executor ./cmd/etcd-mesos-executor
+
+.PHONY: build_proxy
+.SILENT: build_proxy
+build_proxy: bin
+	echo "Building etcd-mesos-proxy.."
+	go build -o bin/etcd-mesos-proxy ./cmd/etcd-mesos-proxy
+
+.PHONY: build_etcd
+.SILENT: build_etcd
+build_etcd: bin
+	echo "Building etcd binaries.."
+	@git submodule init
+	@git submodule update
 	cd _vendor/coreos/etcd; ./build; mv bin/* ../../../bin/
```

Comment:
maybe as a follow-up ticket we can examine the usefulness of having both `_vendor` and `vendor`